en/serialization.md

```diff
@@ -33,3 +31,54 @@ Whatever language you are using, the resulting binary data will be the same:
 <rollspeed> <pitchspeed> <yawspeed> <crc1> <crc2>
 ```
 {% endmethod %}
+
+
+## Field Reordering and CRC Extra Calculation
+
+MAVLink uses one extra CRC that is added to the message CRC to detect mismatches in message specifications. This is to prevent two devices using different message versions from incorrectly decoding a message with the same length.
+
+
+### Rationale for a Format Checksum
+
+While MAVLink 0.9 was used there were a small number of incidents where the XML describing a message that was in active use changed. The change was such that the length of the message didn't change, but the fields did. Revision 0.9 did check the correct message length, but not the field order, types or field names. This meant that, when a MAV using the old code was talking to a ground station using updated code, the fields were badly corrupted. The MAVLink 0.9 protocol completely relied on everyone being careful not to change the meaning or format of any existing message. With so many people working on MAVLink this was hard to enforce. So for MAVLink 1.0 this problem was solved by adding a 1 byte 'seed' to the checksum based on the XML for the message.
+
+
+### CRC_EXTRA Calculation
+
+When the MAVLink code generator runs, it takes a checksum of the XML structure for each message and creates an array define `MAVLINK_MESSAGE_CRCS`. This is used to initialise the `mavlink_message_crcs[]` array in the C/C++ implementation, and is similarly used in the python (or any other, such as the C# and JavaScript) implementation. When the checksum for a message is calculated, this extra byte is added on the end of the data that the checksum is calculated over. The result is that if the XML changes then the message will be rejected by the recipient as having an incorrect checksum. This ensures that only messages where the sender and recipient are using the same message structure will get through (or at least it makes a mistake much more unlikely, as for any checksum application). If you are doing your own implementation of MAVLink 1.0 you can get this checksum in one of two ways: you can just use the generated headers, and use `MAVLINK_MESSAGE_CRCS` to get the right seed for each message type or you can re-implement the code that calculates the seed.
```

Comment:
Is this the CRC2 mentioned in the output in previous section:
---
Whatever language you are using, the resulting binary data will be the same:
```
0x55 0x1C 0x1E <time> <roll> <pitch> <yaw>
<rollspeed> <pitchspeed> <yawspeed> <crc1> <crc2>
```

Reply:
I'm pretty sure, that crc1/2 is bytes of CRC16 checksum. Because extra crc byte calculated from message definition.

Reply:
CRC1/2 are the low/high bytes of the CRC16 calculation. During calculation of the CRC16, it uses corresponding "CRC_EXTRA" for that particular message, which is like a magic byte. Hope that helps.

Reply:
Yes it does help thanks guys - essentially it is part of the calculation, but not something that would appear in the output.